pennylane/gradients/finite_difference.py

```diff
@@ -22,117 +22,155 @@
 import pennylane as qml
 
 
-def finite_diff_stencil(n, order, form):
-    """Generate the finite difference stencil (shift and coefficients)
-    for various derivatives, accuracy, and form.
+def finite_diff_coeffs(n, approx, strategy):
+    r"""Generate the finite difference shift values and corresponding
+    term coefficients for various derivative order, approximation accuracy,
+    and strategy.
 
     Args:
-        n (int): Positive integer specifying the derivative. ``n=1``
-            corresponds to the first derivative.
-        order (int): Positive integer referring to the accuracy of the
-            returned stencil. E.g., ``order=1`` corresponds to a stencil
-            that returns a first-order approximation to the derivative.
-        form (str): one of ``"forward"``, ``"center"``, or ``"backward"``.
+        n (int): Positive integer specifying the order of the derivative. For example, ``n=1``
+            corresponds to the first derivative, ``n=2`` the second derivative, etc.
+        approx (int): Positive integer referring to the approximation order of the
+            returned coefficients. E.g., ``approx=1`` corresponds to the
+            first-order approximation to the derivative.
+        strategy (str): One of ``"forward"``, ``"center"``, or ``"backward"``.
+            For the ``"forward"`` strategy, the finite-difference shifts occur at the points
+            :math:`x_0, x_0+h, x_0+2h,\dots`, where :math:`h` is some small
+            stepsize. The ``"backwards"`` strategy is similar, but in
+            reverse: :math:`x_0, x_0-h, x_0-2h, \dots`. Finally, the
+            ``"center"`` strategy results in shifts symmetric around the
+            unshifted point: :math:`\dots, x_0-2h, x_0-h, x_0, x_0+h, x_0+2h,\dots`.
 
     Returns:
-        array[float]: a ``(2, N)`` array. The first row corresponds to the
+        array[float]: A ``(2, N)`` array. The first row corresponds to the
         coefficients, and the second row corresponds to the shifts.
 
     **Example**
 
-    >>> finite_diff_stencil(n=1, order=1, form="forward")
+    >>> finite_diff_coeffs(n=1, approx=1, strategy="forward")
     array([[-1.,  1.],
            [ 0.,  1.]])
-    >>> finite_diff_stencil(n=1, order=2, form="center")
+
+    The first row corresponds to the coefficients, and the second corresponds
+    to the shifts. For example, this results in the linear combination:
+
+    .. math:: \frac{-y(x_0) + y(x_0 + h)}{h}
+
+    where :math:`h` is the finite-difference step-size.

```

Comment:
```suggestion
    where :math:`h` is the finite-difference step size.
```

Not sure what is the correct spelling of "step size", but just a note to normalize throughout.